src/debugpy/common/json.py

```diff
@@ -106,6 +108,15 @@ def of_type(*classinfo, **kwargs):
     def validate(value):
         if (optional and value == ()) or isinstance(value, classinfo):
             return value
+        elif (
+            isinstance(value, str)
+            and all(x in string.digits + "." for x in value)
```

Comment:
Might as well skip this check, and instead just fall through to the error branch below if neither conversion works.